### PR TITLE
feat(deferred/deferredresult): add deferred.map, deferredresult.map, and deferredresult.mapErr

### PR DESCRIPTION
**Pull request by rmjohnson**

### CODE REVIEW

File: src/Deferred/Deferred.ts [hunk 1]
```diff
@@ -248,3 +248,26 @@ export const isResolvedWith = <A>(
...
+ * @group Mapping
+ *
+ * @example
+ * pipe(
+ *     Deferred.resolved("cheese"),
+ *     Deferred.map(s => s.length),
+ *     Deferred.matchOrElse({
+ *         resolved: r => r,
+ *         orElse: 0
+ * 	    })
```
Comment (taylorn16) [inline src/Deferred/Deferred.ts]: ⛏️ 
```suggestion
 *     Deferred.matchOrElse({
 *         resolved: r => r,
 *         orElse: 0
 *     })
```

File: src/DeferredResult/DeferredResult.spec.ts [hunk 2]
```diff
@@ -277,4 +277,73 @@ describe("DeferredResult", () => {
             }
         )
     })
+
+    describe("map", () => {
+        it("returns a Resolved with projected inner value if Resolved and Ok", () => {
+            const incr = (n: number) => n + 1
+            expect(
+                pipe(Deferred.resolved(Result.ok(22)), DeferredResult.map(incr))
+            ).toStrictEqual(Deferred.resolved(Result.ok(23)))
```
Comment (taylorn16) [inline src/DeferredResult/DeferredResult.spec.ts]: ❓ seeing this makes me wonder whether we could also easily add constructor functions for `DeferredResult.ok` and `DeferredResult.err`?

File: src/DeferredResult/DeferredResult.ts [hunk 3]
```diff
@@ -154,3 +154,51 @@ export const matchOrElse =
...
+ *     Deferred.resolved(Result.ok("cheese")),
+ *     DeferredResult.map(s => s.length),
+ *     DeferredResult.matchOrElse({
+ *         resolvedOk: r => r,
+ *         orElse: 0
+ * 	    })
+ * ) // => 6
+ */
+export const map = <A, E, B>(f: (a: A) => B) =>
+    match<A, E, DeferredResult<B, E>>({
```
Comment (taylorn16) [inline src/DeferredResult/DeferredResult.ts:193]: excellent use of `match` 🚀

File: src/DeferredResult/DeferredResult.ts [hunk 4]
```diff
@@ -154,3 +154,51 @@ export const matchOrElse =
...
+ * @group Mapping
+ *
+ * @example
+ * pipe(
+ *     Deferred.resolved(Result.ok("cheese")),
+ *     DeferredResult.map(s => s.length),
+ *     DeferredResult.matchOrElse({
+ *         resolvedOk: r => r,
+ *         orElse: 0
+ * 	    })
```
Comment (taylorn16) [inline src/DeferredResult/DeferredResult.ts]: ⛏️ on the tab/spacing here as well in this example

File: src/DeferredResult/DeferredResult.spec.ts [hunk 5]
```diff
@@ -277,4 +277,73 @@ describe("DeferredResult", () => {
             }
         )
     })
+
+    describe("map", () => {
+        it("returns a Resolved with projected inner value if Resolved and Ok", () => {
+            const incr = (n: number) => n + 1
```
Comment (taylorn16) [inline src/DeferredResult/DeferredResult.spec.ts:299]: 🎨❓ do we need the `incr` binding in the other tests below? I see how it aids test readability here, but in the test cases where it's not used, we could just inline it, right? (similar question about the `Deferred` tests above)

**Review by taylorn16 — COMMENTED** [review]
Thanks for doing this!! Looks good, just a few nitpicks but we can go as-is if you'd like

File: src/DeferredResult/DeferredResult.spec.ts (hunk 5 above)
Comment (taylorn16) [reply src/DeferredResult/DeferredResult.spec.ts:299]: https://github.com/fp-toolkit/fp-toolkit/pull/66/commits/56461bc3042a4c0f1f28ced77ad111edc3816758

File: src/DeferredResult/DeferredResult.ts (hunk 4 above)
Comment (taylorn16) [reply src/DeferredResult/DeferredResult.ts]: https://github.com/fp-toolkit/fp-toolkit/pull/66/commits/56461bc3042a4c0f1f28ced77ad111edc3816758

File: src/DeferredResult/DeferredResult.spec.ts (hunk 2 above)
Comment (taylorn16) [reply src/DeferredResult/DeferredResult.spec.ts]: https://github.com/fp-toolkit/fp-toolkit/pull/66/commits/56461bc3042a4c0f1f28ced77ad111edc3816758

File: src/Deferred/Deferred.ts (hunk 1 above)
Comment (taylorn16) [reply src/Deferred/Deferred.ts]: https://github.com/fp-toolkit/fp-toolkit/pull/66/commits/56461bc3042a4c0f1f28ced77ad111edc3816758

File: src/Deferred/Deferred.ts [hunk 6]
```diff
@@ -248,3 +248,26 @@ export const isResolvedWith = <A>(
...
+ * @group Mapping
+ *
+ * @example
+ * pipe(
+ *     Deferred.resolved("cheese"),
+ *     Deferred.map(s => s.length),
+ *     Deferred.matchOrElse({
+ *         resolved: r => r,
+ *         orElse: 0
+ * 	   })
```
Comment (rmjohnson-olo) [inline src/Deferred/Deferred.ts]: gah.. I messed up spacing here =(

Comment (olo-ntaylor) [reply src/Deferred/Deferred.ts]: https://github.com/fp-toolkit/fp-toolkit/pull/66/commits/e4530f107c0c5836e5e530b12ffd0b71c7d39e3d perhaps??

**Review by taylorn16 — APPROVED** [review]

**Pull request merged**